### PR TITLE
Feature/946 new disposition states

### LINKED ISSUE #946: New disposition states
ncd - desired disposition diagram

### PR DESCRIPTION
**Pull request by nekron**
Fixes #946

### CODE REVIEW

File: lib/ask/runtime/broker.ex [hunk 1]
```diff
@@ -164,8 +165,19 @@ defmodule Ask.Runtime.Broker do
   end
 
   defp set_stalled_respondents_as_failed(survey) do
+    from(r in assoc(survey, :respondents), where: r.state == "stalled" and r.disposition == "queued")
```
Comment (fgasperij) [inline lib/ask/runtime/broker.ex]: Is the disposition history needed in this cases? Is it being created?

File: lib/ask/runtime/broker.ex [hunk 2]
```diff
@@ -303,15 +316,20 @@ defmodule Ask.Runtime.Broker do
...
 
   defp sync_step_internal(session, reply, session_mode) do
-    respondent = session.respondent
-
     transaction_result = Repo.transaction(fn ->
       try do
+        session = if reply == Flow.Message.answer() do
+          update_respondent_disposition(session, "contacted")
+        else
+          update_respondent_disposition(session, "started")
```
Comment (fgasperij) [inline lib/ask/runtime/broker.ex]: If a respondent jumps from 'queued' to 'started', do the 'contacted' disposition history be created anyway?

File: test/lib/runtime/broker_test.exs [hunk 3]
```diff
@@ -504,35 +571,184 @@ defmodule Ask.BrokerTest do
...
+    survey = Repo.get(Survey, survey.id)
+    assert survey.state == "completed"
+    respondent = Repo.get(Respondent, respondent.id)
+    assert respondent.state == "failed"
+    assert respondent.disposition == "unresponsive"
+
+    :ok = broker |> GenServer.stop
+  end
+
+  test "stalled respondents are marked failed after the survey is completed" do
```
Comment (fgasperij) [inline test/lib/runtime/broker_test.exs]: 'When uncontacted'

File: lib/ask/runtime/broker.ex (hunk 2 above)
Comment (nekron) [inline lib/ask/runtime/broker.ex]: No, it's not being created.

That condition will only be met if the delivery notification of an sms is missed. And only for that mode.

I think that having this missing transition is clearly an indication of something being wrong somewhere. I wouldn't fake it. I'd rather use it to try to find the problem.

File: lib/ask/runtime/broker.ex (hunk 1 above)
Comment (nekron) [inline lib/ask/runtime/broker.ex]: Actually, the disposition shouldn't change in this case. Sending an additional commit to fix this.

File: test/lib/runtime/broker_test.exs (hunk 3 above)
Comment (nekron) [inline test/lib/runtime/broker_test.exs]: Fixed

**Pull request merged**